exporter/coralogixexporter/client.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2021, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package coralogixexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/coralogixexporter"
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"os"
+
+	cxpb "github.com/coralogix/opentelemetry-cx-protobuf-api/coralogixpb"
+	"go.opentelemetry.io/collector/config/configgrpc"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.opentelemetry.io/collector/model/pdata"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/metadata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+)
+
+// CoralogixClient by Coralogix
+type CoralogixClient struct {
+	cfg      Config
+	logger   *zap.Logger
+	cfggrpc  configgrpc.GRPCClientSettings
+	conn     *grpc.ClientConn
+	client   cxpb.CollectorServiceClient
+	metadata cxpb.Metadata
+}
+
+// NewCoralogixClient by Coralogix
+func NewCoralogixClient(cfg *Config, logger *zap.Logger) *CoralogixClient {
+	// check env variables
+	endpoint := cfg.Endpoint
+	privateKey := cfg.PrivateKey
+	appName := cfg.AppName
+	subSystem := cfg.SubSystem
+
+	if os.Getenv("CORALOGIX_ENDPOINT") != "" {
```

Comment:
I would really avoid dealing with env vars here. Users can use env vars in the config, and this PR causes this component to deviate from other components.